pymatgen/io/lobster/inputs.py

```diff
@@ -160,21 +160,21 @@ def __setitem__(self, key, val: Any) -> None:
 
         super().__setitem__(key, val.strip() if isinstance(val, str) else val)
 
-    def __getitem__(self, key) -> Any:
+    def __getitem__(self, key: str) -> Any:
         """To avoid cases sensitivity problems."""
         try:
-            return self.data[key.strip().lower()]
+            return super().__getitem__(key.strip().lower())
```

Comment:
I'm not sure if we have a reason to bypass the method resolution order (especially when we chose to inherit from `UserDict` here), let me know if there's anything that I'm not aware of :)

Reply:
I am not sure what you mean with bypass here?

Reply:
The original implementation (access directly through the internal dict `self.data` would bypass the MRO), I have changed it to access through the `__getitem__` magic method to respect this.